docs/src/prototypes/alerts/AnimatedBannerAlert.tsx

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react'
+import { Alert, Animation, SiteVariablesInput, AlertProps } from '@stardust-ui/react'
+
+interface BannerAlertProps extends AlertProps {
+  oof?: boolean
+  critical?: boolean
+  urgent?: boolean
+}
+
+/**
+ * Customized Alert that can express different intents through boolean props:
+ * oof, urgent, critical
+ */
+const BannerAlert: React.FunctionComponent<BannerAlertProps> = props => {
+  const { oof, critical, urgent, ...rest } = props
+  return (
+    <Alert
+      variables={(siteVars: SiteVariablesInput) => {
+        const appWhite = siteVars.colors.grey[50]
+
+        if (oof) {
+          return {
+            color: siteVars.orchid,
+            backgroundColor: '#F9F5F8',
+            borderColor: '#DCADC7',
+          }
+        }
+
+        if (urgent) {
+          return {
+            color: appWhite,
+            backgroundColor: siteVars.red,
+            borderColor: siteVars.red,
+          }
+        }
+
+        if (critical) {
+          return {
+            color: appWhite,
+            backgroundColor: siteVars.gray03,
+            borderColor: siteVars.gray02,
+          }
+        }
+
+        return {}
+      }}
+      {...rest}
+    />
+  )
+}
+
+interface AnimatedBannerAlertProps extends BannerAlertProps {
+  closable?: boolean
+}
+
+interface AnimatedBannerAlertState {
+  open: boolean
+}
+
+/**
+ * Needs to have 'slideDown' animation defined in parent Provider
+ */
+class AnimatedBannerAlert extends React.Component<
+  AnimatedBannerAlertProps,
+  AnimatedBannerAlertState
+> {
+  private readonly initialState: AnimatedBannerAlertState = { open: true }
+
+  state = this.initialState
+
+  componentDidUpdate(prevProps: AnimatedBannerAlertProps) {
+    const p = prevProps
+    const { info, danger, oof, critical, urgent } = this.props
+
+    if (
+      p.info !== info ||
+      p.danger !== danger ||
+      p.oof !== oof ||
+      p.critical !== critical ||
+      p.urgent !== urgent
+    ) {
+      this.setState(this.initialState)
+    }
+  }
```

Comment:
Instead of manual reset, we can use `key: bannerName` in `BannerAlerts.tsx`. This will remount component and will reset state, so we will be able to remove `componentDidUpdate()` completely.

Reply:
done, great suggestion 💯